optimum/intel/openvino/quantization.py

```diff
@@ -645,14 +666,6 @@ def _weight_only_quantization(
             dataset = calibration_dataset
         else:
             dataset = nncf.Dataset(calibration_dataset)
-    elif config.dataset is not None and isinstance(config.dataset, str):
-        tokenizer = AutoTokenizer.from_pretrained(config.tokenizer)
-
-        from optimum.gptq.data import get_dataset, prepare_dataset
-
-        nsamples = config.num_samples if config.num_samples else 128
-        dataset = get_dataset(config.dataset, tokenizer, seqlen=32, nsamples=nsamples)
-        dataset = prepare_dataset(dataset)
```

Comment:
`_weight_only_quantization` is still used [here](https://github.com/huggingface/optimum-intel/blob/b383ffb2539b704429a8b830dc0ed0bd68703d8e/optimum/intel/openvino/modeling_base.py#L142) and [here](https://github.com/huggingface/optimum-intel/blob/b383ffb2539b704429a8b830dc0ed0bd68703d8e/optimum/exporters/openvino/convert.py#L80) we might need to update this places as well to ensure compatibility

Reply:
Yep, I did want to do that, but the difference there is that only a raw `openvino.runtime.Model` is available, not an instance of `transformers.PreTrainedModel`, and the latter is required to initialize `OVQuantizer`. We could extend `OVQuantizer` to accept an instance of `openvino.runtime.Model`, but that's a rather serious API change.